Assets/Scripts/BulletManagement/Bullet.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+
+/**
+ * Base class for bullets. Bullets are destroyed when they no longer are
+ * in the view of the Camera.
+ */
+public class Bullet : MonoBehaviour
+{
+    public float decayTime = 1f;
+    public float instantiationTime = 0.0f;
+
+    // Metadata
+    public Emitter emitter = null;
+    public BulletPattern bulletPattern = null;
+    private int bulletId;
+
+    void Start()
+    {
+        // Destroy self in N seconds after emitt
+        // Do not do so if set to zero.
+        if (decayTime != 0)
+        {
+            Destroy(this.gameObject, decayTime);
+        }
+        this.instantiationTime = Time.timeSinceLevelLoad;
+    }
+
+    void OnBecameInvisible() {
+        Destroy(this.gameObject);
+    }
+
+    void Update()
+    {
+        float timeAlive = Time.timeSinceLevelLoad - this.instantiationTime;
+
+        // Move in a fixed direction until otherwise
+        // Optimize this call if need be when we hit optimization issues.
+        Vector3 translation = this.bulletPattern.GetTranslation(timeAlive, this.bulletId) * Time.deltaTime;
```

Comment:
Question: since `timeAlive` is always increasing, won't these bullets be continually accelerating since we're scaling `translation` by an ever-increasing amount? I would think we'd want to translate it relative to its last known location rather than relative to its point of origin.

Reply:
Good catch. Get Translation should return derivative of position function.